### PR TITLE
Webpack defaults

### LINKED ISSUE #8: Use Travis for CI
Once #3 has been implemented, we should use Travis to automatically run tests for all PRs. In addition, Travis can automatically handle npm publishing for us.

### PR DESCRIPTION
**Pull request by ChadKillingsworth**
Add the webpack-defaults to the project.

Closes #8

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      env: WEBPACK_VERSION=latest JOB_PART=coverage
+      script: npm run travis:$JOB_PART
+      after_success: 'bash <(curl -s https://codecov.io/bash)'
+    - stage: Webpack canary
+      before_script: npm i --no-save git://github.com/webpack/webpack.git#master
+      script: npm run travis:$JOB_PART
+      node_js: 8
+      env: WEBPACK_VERSION=canary JOB_PART=test
+before_install:
+  - 'if [[ `npm -v` != 5* ]]; then npm i -g npm@^5.0.0; fi'
```
Comment (d3viant0ne) [inline .travis.yml]: `'if [[ `npm -v` != 5* ]]; then npm i -g npm@latest; fi'`

Should fix the problem

**Comment by d3viant0ne** [comment]
This has been failing because npm & yarn don't play well together.

Delete your yarn.lock so travis doesn't try & use it

**Comment by d3viant0ne** [comment]
I'm sure there is a more elegant way to tell Travis to ignore a yarn.lock file, I just never bothered to figure our what that is.

File: appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      webpack_version: latest
+      job_part: test
+    - nodejs_version: '4.3'
+      webpack_version: latest
+      job_part: test
+build: 'off'
+matrix:
+  fast_finish: true
+install:
+  - 'ps: Install-Product node $env:nodejs_version x64'
```
Comment (d3viant0ne) [inline appveyor.yml]: Strip the ` ' ` off of the shell commands.

Comment (d3viant0ne) [inline appveyor.yml]: Not entirely sure when that started but it shouldn't & it won't in a minute

**Comment by ChadKillingsworth** [comment]
Alright - travis is finally passing. @d3viant0ne are you ok with these changes from defaults?

**Comment by d3viant0ne** [comment]
shipit

**Pull request merged**